Core/Object Arts/Dolphin/IDE/Base/InternalIcon.cls

```diff
@@ -182,12 +182,55 @@ imageManager
 	^IconImageManager current!
 
 new
-	^self withExtent: self defaultExtent! !
+	^self withExtent: self defaultExtent!
+
+stbConvert: anArray fromVersion: anInteger 
+	"Private - Convert from earlier version view by updating and answering the array of instance
+	variables, instVarArray. "
+
+	| instVars |
+	instVars := anArray.
+	anInteger < 1 ifTrue: [instVars := self stbConvertFromVersion0: instVars].
+	^instVars!
+
+stbConvertFrom: anSTBClassFormat 
```

Comment:
This will work, but Model has a fairly complex upgrade requirements. For simple cases like this, the pattern used in ImageManager will be adequate:
```
stbConvertFrom: anSTBClassFormat 
	"Version 1 adds backcolor and maskcolor instance variables."

	^
	[:data | 
	| newInstance |
		newInstance := self basicNew.
		1 to: data size do: [:i | newInstance instVarAt: i put: (data at: i)].
		newInstance]
```
This form will always work if adding inst vars at the end that can be left nil.
I'll merge it anyway to see if the test passes.